e2e_test/batch/aggregate/duckdb/test_minmax.slt.part

```diff
@@ -0,0 +1,24 @@
+# Copied from https://github.com/duckdb/duckdb (MIT licensed).
+# Copyright 2018-2022 Stichting DuckDB Foundation
+
+statement ok
+CREATE TABLE lists(l int[]);
+
+statement ok
+INSERT INTO lists VALUES ({0, 10}), ({1, 11}), ({2, 12}), ({3, 13}), ({4, 14}), ({5, 15}); 
```

Comment:
I fail this query on postgres. I guess it's a duckdb-specific syntax?

Reply:
```sql
INSERT INTO lists VALUES ('{0, 10}'), ('{1, 11}'), ('{2, 12}'), ('{3, 13}'), ('{4, 14}'), ('{5, 15}');
```

This syntax works. The duckdb one will break our compatibility with postgres, though I do prefer this syntax.

Reply:
risingwave supports creating array using 'array[1, 2, 3]' or '{1, 2, 3}' and pg does not support the latter...

Reply:
Ok yes indeed the syntax should be compatible with pg, will fix.

Reply:
> ```sql
> INSERT INTO lists VALUES ('{0, 10}'), ('{1, 11}'), ('{2, 12}'), ('{3, 13}'), ('{4, 14}'), ('{5, 15}');
> ```
> 
> This syntax works. The duckdb one will break our compatibility with postgres, though I do prefer this syntax. The postgres way is like deferring the parsing phase to string casting.

Solution for accepting single-quoted '{1, 2, 3}' arrays (like pg does) as opposed to {1, 2, 3} like we currently do. 

Three scenario's:
- `ARRAY[1]` should be parsed as `integer[]`: completed
- `{1}` syntax without single quotes: is now deprecated in this PR
- `'{1}'` pg-like syntax: is now implemented in this PR

The problem with `'{1}'` style syntax is that it is ambiguous for the parser: it is a valid `Varchar` value as well as a valid `integer[]` value. 

That's why `'{1}'` will be parsed as a SingleQuoteString and bound to a `Literal::new(Some(ScalarImpl::Utf8(s)), DataType::Varchar)`. Depending on the context this value can now be implicitly/assign/explicitly casted to a `List`. 

Most relevant e2e tests can be found in `e2e_test/batch/basic/array.slt.part` under `## Test single quote style array syntax.`.